spyderlib/widgets/dicteditor.py

```diff
@@ -1286,8 +1286,6 @@ def setup(self, data, title='', readonly=False, width=500,
         self.resize(width, height)
         
         self.setWindowTitle(self.widget.get_title())
-        if is_text_string(icon):
-            icon = get_icon(icon)

```

Comment:
setup() could be passed a string, (an icon filename). Now it is directly passed a QIcon.